### ISSUE TITLE
Example Jupyter Notebook 'shared_interest_example.ipynb' with  ImageNet Large Scale Visual Recognition Challenge (ILSVRC) data

### ISSUE THREAD

**Issue by dbl001** [mitvis/shared-interest#1]
I downloaded ImageNet data from this Kaggle Challenge - ImageNet Large Scale Visual Recognition Challenge (ILSVRC):

https://www.kaggle.com/competitions/imagenet-object-localization-challenge/data

The labels (WordNet Ids) are all preceded with the character 'n' which I stripped off.
E.g. - 
```
labels = list(map(lambda n: re.sub('[^0-9]', '', n), labels))
labels = np.array(labels)
```
In the download, there are 1,282,169 images and 545,548 annotations.

How do you recommend I handle the missing annotation files?

E.g. like this?

https://github.com/pytorch/pytorch/issues/1137

https://discuss.pytorch.org/t/how-to-skip-wrong-document-in-getitem-in-dataset-class/110659

<img width="908" alt="Screen Shot 2022-07-16 at 9 38 25 AM" src="https://user-images.githubusercontent.com/3105499/179364274-17b95f93-f088-47d4-91dc-1d5a7abee8ba.png">

**Comment by aboggust** [comment]
Hi @dbl001 --- thanks for asking!

That file structure looks good to me! The Shared Interest `ImageNet Dataset` just requires a 1:1 mapping between the images and annotations such that they have the same filename and immediate parent directory name. You have that since `.../n01234567/n01234567_89.JPEG` maps to `.../n01234567/n01234567_89.xml`. You can even keep the 'n' characters if you wanted.

With respect to handling the missing file annotations, either of the solutions you linked to will work. If you only plan to use this data to do Shared Interest analysis, I would recommend preprocessing your data and removing images that do not have annotations. That way you won't have to add error handling and it should speed up you dataloader. If you plan to train models on the data and do not want to remove images, then I would suggest running Shared Interest on the validation set. All 50K images in the ImageNet validation set have associated annotations, and Shared Interest will run much faster on that smaller subset.

Let me know if you have any other questions!

**Comment by dbl001** [comment]
Where do you get the 'label'? 
E.g. -  In the Jupyter notebook 'shared_interest_example.ipynb' the ostrich is 9 and the eagle is 22.

My LOC_synset_mapping.txt:

n01518878 ostrich, Struthio camelus

My train_loc.txt:

 n01518878/n01518878_10004 4635

My LOC_train_solution.csv:
n01518878 347 187 482 371

This appears to be the mapping that corresponds to your 'shared_interest_example.ipynb' example :

https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a

and this:

https://discuss.pytorch.org/t/imagenet-classes/4923/6

**Comment by aboggust** [comment]
`labels` are the ImageNet class labels. They are contained in the `LOC_synset_mapping.txt` from the Kaggle download you are using. In that file, each index maps to an nid and a class name. For instance, the 9th index maps to `n01518878 ostrich, Struthio camelus`.

**Comment by aboggust** [comment]
Hi dbl001, I hope you were able to get Shared Interest up and running!

We have added scripts that help reformat an ImageNet download into PyTorch's ImageFolder file structure. See [imagenet_download_util](https://github.com/mitvis/shared-interest/tree/main/imagenet_download_util) for more details.

If you have any other questions, please open a new issue.

**Comment by aboggust** [comment]
We have included scripts that help reformat an ImageNet download into PyTorch's ImageFolder file structure. See [imagenet_download_util](https://github.com/mitvis/shared-interest/tree/main/imagenet_download_util).

**Comment by dbl001** [comment]
Cool. Thanks for letting me know.
